Sources/SwiftFormatConfiguration/Configuration.swift

```diff
@@ -189,6 +195,25 @@ public struct Configuration: Codable, Equatable {
       forKey: .lineBreakAroundMultilineExpressionChainComponents)
     try container.encode(rules, forKey: .rules)
   }
+
+  /// Configuration file associated with a swift file.
+  /// 
+  /// Looks for a ".swift-format" file in the same directory as the swift file, or its nearest parent.
+  /// If one is not found, returns "nil".
+  public static func configurationFile(forSwiftFile url: URL) -> URL? {
```

Comment:
If we're generalizing this so that other people can call it, we don't have to require that the input URL point specifically to a Swift file. For example, we might want to allow someone to pass the URL of a directory to ask "what configuration file applies to files in this directory?"

With that in mind, we could name this `func url(forConfigurationFileApplyingTo url: URL) -> URL?`, and update the doc comment accordingly, such as

```
/// Returns the URL of the configuration file that applies to the given file or directory.
```

I think we can be vague about the exact search behavior in the doc comment because it will give us the flexibility to adjust it in the future as well.

But we'll also need to update the implementation below: before the loop, we'll need to check to see if the incoming URL is a file (rather than a directory), and if it is, delete the file component from the path. Then we'll also need to adjust the first line of the loop and the condition slightly to make sure that we still go up to the parent at the right time and exit correctly when we hit the root.

Reply:
I added a placeholder component for files instead of adjusting the loop condition, because otherwise it would be hard to write it in a way that doesn't skip the root directory